.github/workflows/velox_be.yml

```diff
@@ -631,3 +616,17 @@ jobs:
         if: ${{ always() }}
         run: |
           $PATH_TO_GLUTEN_TE/$OS_IMAGE_NAME/gha/gha-checkout/clean.sh
+
+  upload-golden-file:
+    runs-on: velox-self-hosted
+    needs: [ubuntu2004-test-spark32, ubuntu2004-test-spark33, ubuntu2004-test-spark34]
+    steps:
+      - name: Upload golden files
```

Comment:
if: failure()

Reply:
Right, we should trigger this upload job when the previous job fail.

And according to [Github Action Doc](https://docs.github.com/en/actions/learn-github-actions/expressions#failure), we should add this condition on job instead of adding on each step.